query/physicalplan/aggregate.go

```diff
@@ -390,7 +410,7 @@ func (a *HashAggregate) Finish(ctx context.Context) error {
 	defer span.End()
 
 	numCols := len(a.groupByCols) + 1
-	numRows := len(a.arraysToAggregate)
+	numRows := len(a.columns[0].arrays)
```

Comment:
Will the length of `arrays` be equal across all `columns[i]`? This code is starting to get pretty complex/hard to understand but maybe it's just me.

Reply:
Yeah, ask me about implementing these things. I guess we should figure out a way to refactor things into smaller pieces. An attempt was the introduction of `AggregateColumn`.

Reply:
> Will the length of `arrays` be equal across all `columns[i]`?

I double-checked this. This way of getting the num of Rows is fine.
For each different hash, we group by we append a new array for each aggregation to this array. Each of these arrays is going to be aggregated into a single value based on its function.
They `len(arrays)` will be consistent across aggregations.

Here is the code where this is taken care of:
https://github.com/polarsignals/frostdb/blob/4d6fe5bb56bf0d4527b25691c509ee977202601a/query/physicalplan/aggregate.go#L364-L367

Reply:
To describe this with a relationship. Each aggregation will have the exact same amount of arrays, because they each correspond to a hash in the `hashToAggregate` map.

Reply:
I'll add this as a comment though, so it's less confusing in the future! :+1:

Reply:
Actually, this should be way more descriptive.
```
// Each hash that's aggregated by will become one row in the final result.
numRows := len(a.hashToAggregate)
```